copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopyBookRecordReader.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch;
+
+import net.sf.JRecord.Common.AbstractFieldValue;
+import net.sf.JRecord.Details.AbstractLine;
+import net.sf.JRecord.External.Def.ExternalField;
+import net.sf.JRecord.External.ExternalRecord;
+import net.sf.JRecord.IO.AbstractLineReader;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import static co.cask.hydrator.plugin.batch.CopybookSource.COPYBOOK_INPUTFORMAT_CBL_CONTENTS;
+import static co.cask.hydrator.plugin.batch.CopybookSource.COPYBOOK_INPUTFORMAT_DATA_HDFS_PATH;
+import static co.cask.hydrator.plugin.batch.CopybookSource.COPYBOOK_INPUTFORMAT_FILE_STRUCTURE;
+import static co.cask.hydrator.plugin.batch.CopybookSource.DEFAULT_FILE_STRUCTURE;
```

Comment:
this is a sign that these constants should not be in the plugin, but should be in this class.